vertical-pod-autoscaler/pkg/admission-controller/resource/vpa/matcher_test.go

```diff
@@ -52,7 +70,7 @@ func TestGetMatchingVpa(t *testing.T) {
 			name: "matching selector",
 			pod:  podBuilder.Get(),
 			vpas: []*vpa_types.VerticalPodAutoscaler{
-				vpaBuilder.WithUpdateMode(vpa_types.UpdateModeAuto).WithName("auto-vpa").Get(),
+				vpaBuilder.WithUpdateMode(vpa_types.UpdateModeAuto).WithName("auto-vpa").WithTargetRef(targetRef).Get(),
```

Comment:
Please correct me if I am wrong. I thought the target ref of vpa is usually "Deployment" kind. E.g., https://github.com/kubernetes/autoscaler/tree/master/vertical-pod-autoscaler#example-vpa-configuration

Are we assigning `ReplicationController` kind here?

Reply:
The test is using a `controllerfetcher.FakeControllerFetcher` to retrieve the owner reference. The implementation of this mock returns the same reference that is passed. So for the test it is important that the reference set in the pod is the same that the one set in the VPA. We are left with 2 options, none reflecting really the reality, and I would say it is not important because here we are not testing the code of the embedded `controllerfetcher`:
Option 1: vpa.targetRef=> Deployment    and pod.ownerRef=> Deployment (which is not possible in reality)
Option 2: pod.ownerRef=> ReplicaSet    and pod.ownerRef=> Replicaset (which is not happening in reality)

In fact there is a 3rd option that we can implement and the would be more correct compare to reality: we use Statefulset. I can amend the PR for that.

Reply:
Just to clarify, the functional code does traverse ownerRefs (pod -> replica-set -> deployment)?
If this test is really not meant to test the owner chain resolution (and if we really have another test making sure that still happens correctly) I'm okay with a surreal test case given it's explained in a comment.

Reply:
https://github.com/kubernetes/autoscaler/pull/6460/commits/4f9f840018686e4444b9812e9871fb79b6f15142